pkg/controller/session/model_convert.go

```diff
@@ -7,7 +7,7 @@ import (
 
 const (
 	// DefaultRouteHeaderName holds the name of the Header used to route traffic if no Route is provided
-	DefaultRouteHeaderName = "X-Workspace-Route"
+	DefaultRouteHeaderName = "x-workspace-route"
```

Comment:
According to istio docs header matchers should be lower case

> The header keys must be lowercase and use hyphen as the separator, e.g. `x-request-id`

Reply:
https://istio.io/docs/reference/config/networking/v1alpha3/virtual-service/#HTTPMatchRequest